packages/site-kit/src/lib/search/search.js

```diff
@@ -49,33 +81,47 @@ export function init(blocks) {
 /**
  * Search for a given query in the existing index
  * @param {string} query
- * @returns {import('./types').Tree[]}
+ * @returns {Promise<import('./types').Tree[]>}
  */
-export function search(query) {
-	const escaped = query.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
-	const regex = new RegExp(`(^|\\b)${escaped}`, 'i');
-
-	const blocks = indexes
-		.flatMap((index) => index.search(query))
-		.map(lookup)
-		.map((block, rank) => ({ block: /** @type{import('./types').Block} */ (block), rank }))
-		.sort((a, b) => {
-			const a_title_matches = regex.test(/** @type {string} */ (a.block.breadcrumbs.at(-1)));
-			const b_title_matches = regex.test(/** @type {string} */ (b.block.breadcrumbs.at(-1)));
-
-			// massage the order a bit, so that title matches
-			// are given higher priority
-			if (a_title_matches !== b_title_matches) {
-				return a_title_matches ? -1 : 1;
-			}
+export async function search(query) {
+	const search_results = /** @type {any[]} */ (
+		(
+			await orama_search(index, {
+				term: query,
+				sortBy: (a, b) => {
+					const [_docIdA, scoreA, docA] = a;
+					const [_docIdB, scoreB, docB] = b;
+
+					// @ts-ignore
+					return docB.priority * 1000 + scoreB - (docA.priority * 1000 + scoreA);
+				},
+				boost: {
+					h1: 3,
+					h2: 2,
+					h3: 1
+				},
+
+				limit: search_appropriate_blocks.length
+			})
+		).hits.map(({ document }) => document)
+	);
+
+	/** @type {import('./types').SearchAppropriateBlock[]} */
+	const blocks = [];
+
+	for (const result of search_results) {
+		// @ts-ignore
```

Comment:
is this `@ts-ignore` needed? it doesn't look like it would be, but if so, then perhaps it should be `@ts-expect-error`?